Bluejay.asm

```diff
@@ -3593,21 +3609,45 @@ decode_temp_done:
 	mov	Temp1, #Pgm_Beep_Strength	; Read programmed beep strength setting
 	mov	Beep_Strength, @Temp1		; Set beep strength
 
-	mov	Temp1, #Pgm_Braking_Strength	; Read programmed braking strength setting
+	; Decode pwm frequency setting
+	mov	Temp1, #Pgm_Pwm_Frequency	; Read programmed pwm frequency setting
 	mov	A, @Temp1
-IF PWM_BITS_H == 3					; Scale braking strength to pwm resolution
-	; Note: Added for completeness
+	mov	Temp2, #0					; 24 kHz by default
+	cjne	A, #48, ($+5)
+	mov	Temp2, #1					; 48 kHz
+	cjne	A, #96, ($+5)
+	mov	Temp2, #2					; 96 kHz
+
+	; TODO: Simplify?
+	clr	C
+	mov	A, #(2 + MCU_48MHZ - PWM_CENTERED)
+	subb	A, Temp2
+	mov	Pwm_Bits_H, A				; Number of bits in pwm high byte
+
+	; Decode pwm dithering and braking strength settings
+	mov	Temp1, #Pgm_Dithering		; Read programmed dithering setting
+	mov	A, @Temp1
+	add	A, #0FFh					; Carry set if A is not zero
+	mov	Flag_Dithering, C			; Set dithering enabled
+
+	mov	A, Pwm_Bits_H
+	cjne	A, #3, decode_pwm_h_2
 	; Currently 11-bit pwm is only used on targets with built-in dead time insertion
```

Comment:
Here Pwm_H_Mask value should be initialized to a correct value, because it is used above